mathesar/models/base.py

```diff
@@ -283,6 +283,60 @@ class Explorations(BaseModel):
     description = models.CharField(null=True)
 
 
+class Form(BaseModel):
+    id = models.UUIDField(primary_key=True)
+    name = models.CharField()
+    version = models.IntegerField()
+    database = models.ForeignKey('Database', on_delete=models.CASCADE)
+    base_table_oid = models.PositiveBigIntegerField()
+    schema_oid = models.PositiveBigIntegerField()
+    description = models.CharField(null=True)
+    header = models.OneToOneField('FormHeader', on_delete=models.DO_NOTHING)
+    submission = models.ForeignKey('SubmissionSettings', on_delete=models.CASCADE)
+    is_public = models.BooleanField(default=False)
+
+
+class FormHeader(BaseModel):
+    title = models.CharField()
+    subtitle = models.CharField(null=True)
+
+
+class SubmissionSettings(BaseModel):
+    role = models.ForeignKey('ConfiguredRole', on_delete=models.CASCADE)
+    message = models.CharField(null=True)
+    redirect_url = models.URLField(null=True)
+    submit_label = models.CharField(null=True)
+
+
+class FormField(BaseModel):
+    id = models.CharField(primary_key=True)
+    attnum = models.SmallIntegerField()
+    table_oid = models.PositiveBigIntegerField()
+    related_form = models.ForeignKey('Form', on_delete=models.CASCADE, related_name='form_field')
+    kind = models.CharField(
+        choices=[
+            ("scalar_column", "scalar_column"),
+            ("foreign_key", "foreign_key"),
+            ("reverse_foreign_key", "reverse_foreign_key")
+        ],
+    )
+    record_summary_template = models.JSONField(null=True)
```

Comment:
Let's remove `record_summary_template` in this PR. I'd like to add it along with `lookup_columns` when the UX for it is confirmed.